### ISSUE TITLE
_test.go files in the test scope.

### ISSUE THREAD

**Issue by squeedee** [go-lang-plugin-org/go-lang-idea-plugin#1905]
Unfortunately IntelliJ is designed to have test roots, not test files. So we can't mark certain files (such as *_test.go) as tests.

And example of why this is painful:

* Open a project
* Navigate to a commonly used Struct
* Find Usages...
* in the usage search pane, select 'Group By test/production'

All the files appear in 'production', and if you are not looking for test usages, you still have to sift through them. There is no easy way to exclude them or ignore them.
![screen shot 2015-10-06 at 11 01 01 am](https://cloud.githubusercontent.com/assets/27838/10312506/8d0b0f96-6c19-11e5-9678-3a81c16eec33.png)

In the following example, i18n.Init is used ONCE in production, which is the usage I was looking for. But I would need to visually comb through 65 usages to find it.

**Comment by dlsniper** [comment]
This is related to https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/1332 (I'll add the comment so that there's a link between issues). Thanks for reporting.

**Comment by dlsniper** [comment]
@ignatov / @zolotov is it safe to assume that flagging files as test files should be done at the platform level? In which case should I open a related issue in YouTrack? I can't seem to find any searching for this feature: https://youtrack.jetbrains.com/issues/IDEA?q=mark+individual+files+as+test (or related searches)

**Comment by zolotov** [comment]
@dlsniper of course nobody will mark all test files as `tests` manually, I don't see any reason for such request

**Comment by dlsniper** [comment]
Makes sense, but like @squeedee said, in Go's case, the tests are in the same directory 99.9% of the time, and are easily identifiable as _test. So I'm thinking that in this case maybe the plugin could flag those files automatically, no? Or should this concern be left only for the find usages functionality?

**Comment by zolotov** [comment]
> So I'm thinking that in this case maybe the plugin could flag those files automatically, no? 

Sure, and that's why this issue is still open

**Comment by dlsniper** [comment]
@zolotov I've tried to find a way to implement this but, at least to my understanding, this is currently impossible to overwrite the `UsageScopeGroupingRule` (from https://github.com/JetBrains/intellij-community/blob/master/platform/usageView/src/com/intellij/usages/impl/rules/UsageScopeGroupingRule.java#L39) to allow defining the own implementation of ` groupUsage `. Am I missing something? Thank you.

**Comment by zolotov** [comment]
@dlsniper this issue requires changes in the platform
